### PR TITLE
Add StatefulSet watcher

### PR DESCRIPTION
**Pull request by sakiib**
Signed-off-by: sakibalamin <sakibalamin@appscode.com>

### CODE REVIEW

File: apis/constants.go [hunk 1]
```diff
@@ -71,3 +71,25 @@ const (
 const (
 	CertificatePath = "/etc/vault/tls"
 )
+
+// Resource kind related constants
+const (
+	ResourceKindStatefulSet = "StatefulSet"
```
Comment (kamolhasan) [inline apis/constants.go:77]: Remove `ResourceKindStatefulSet` & `ResourceKindVaultServer` from here. These variables already exist in their respective API directory. Use those.

File: apis/kubevault/v1alpha1/types.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ const (
 	VaultExporterContainerName = "exporter"
 )
 
-// +kubebuilder:validation:Enum=Provisioning;DataRestoring;Ready;Critical;NotReady;Halted;Sealed;Unsealed;Initializing;Initialized
+// +kubebuilder:validation:Enum=Sealed;Unsealed;Initializing;Initialized
```
Comment (kamolhasan) [inline apis/kubevault/v1alpha1/types.go:29]: Add Ready,Critical,NotReady too

File: apis/kubevault/v1alpha1/vaultserver_types.go [hunk 3]
```diff
@@ -116,14 +116,22 @@ type VaultServerList struct {
 	Items           []VaultServer `json:"items,omitempty" protobuf:"bytes,2,rep,name=items"`
 }
 
-// +kubebuilder:validation:Enum=Processing;Uninitialized;Running;Sealed
+// +kubebuilder:validation:Enum=Initializing;Unsealing;Sealed;Ready;NotReady;Critical
 type ClusterPhase string
```
Comment (kamolhasan) [inline apis/kubevault/v1alpha1/vaultserver_types.go:120]: Use VaultServerPhase instead.
For kubeDB, it is called `clusterPhase`.

File: apis/kubevault/v1alpha1/vaultserver_types.go [hunk 4]
```diff
@@ -116,14 +116,22 @@ type VaultServerList struct {
...
 
 const (
-	ClusterPhaseProcessing    ClusterPhase = "Processing"
-	ClusterPhaseUnInitialized ClusterPhase = "Uninitialized"
-	ClusterPhaseRunning       ClusterPhase = "Running"
-	ClusterPhaseSealed        ClusterPhase = "Sealed"
+	// used for VaultServer that are Initializing
```
Comment (kamolhasan) [inline apis/kubevault/v1alpha1/vaultserver_types.go:123]: Move them to `VaultServerPhase`

**Review by kamolhasan — CHANGES_REQUESTED** [review]

File: apis/constants.go (hunk 1 above)
Comment (sakiib) [reply apis/constants.go]: Looks like only `ResourceKindVaultServer` was already present there, used that. Couldn't find any other occurrence of `ResourceKindStatefulSet`

**Review by kamolhasan — APPROVED** [review]
LGTM!

**Pull request merged**